libcaf_net/src/multiplexer.cpp

```diff
@@ -242,8 +242,10 @@ void multiplexer::shutdown() {
       for (size_t i = 1; i < managers_.size(); ++i)
         write_to_pipe(4, managers_[i]);
     }
+    return true;
   } else {
     write_to_pipe(5, nullptr);
+    return false;
```

Comment:
Why not run `run()` here and return `void` instead?

Reply:
I thought calling `run()` from the `middleman`at the same place as `mpx_->join()` would be clearer code.

Reply:
A shutdown function that returns a boolean that has nothing to do with the success of the operation is not particularly clear to me TBH. 😉

Reply:
I see that and have changed it. :+1: Just out of curiosity: Would this be passable solution if there was type-information in form of an enum or something instead of the bool?

Reply:
Okay wait. @josephnoir just had something to say about this. Your proposed solution would not work. We have to differentiate the two cases and call `run()` from the `middleman` in order to safely shutdown the `multiplexer. I fixed this  and have a commit on the way.

Reply:
Okay wait. @josephnoir just had something to say about this. Your proposed solution would not work. We have to differentiate the two cases and call `run()` from the `middleman` in order to safely shutdown the `multiplexer. I fixed this  and have a commit on the way.